internal/state/execution.go

```diff
@@ -310,29 +310,33 @@ func (blockExec *BlockExecutor) ApplyBlock(
 	return state, nil
 }
 
-func (blockExec *BlockExecutor) ExtendVote(ctx context.Context, vote *types.Vote) (types.VoteExtension, error) {
+func (blockExec *BlockExecutor) ExtendVote(ctx context.Context, vote *types.Vote) ([]byte, error) {
 	req := abci.RequestExtendVote{
-		Vote: vote.ToProto(),
+		Hash:   vote.BlockID.Hash,
+		Height: vote.Height,
```

Comment:
Sanity check here please?

Reply:
From the [current spec](https://github.com/tendermint/tendermint/blob/master/spec/abci%2B%2B/abci++_methods_002_draft.md#extendvote) we only need to supply the header hash (which I'm assuming here is the Merkle root of the header) and the height of the proposed block to extend.

It seems like we leave it up to the application to store information regarding that proposed block either in `PrepareProposal` or `ProcessProposal`.